### PR TITLE
Change anonymous classes to lambdas

### PR DESCRIPTION
**Pull request by res0nance**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

Cleanup the anonymous classes to java 8 lambdas

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]
Looks great! I think there are now some unused imports that need to be removed, and I left some IDE-suggested optimizations as suggestions, but otherwise this looks ready to merge.

File: core/src/main/java/hudson/PluginManager.java [hunk 1]
```diff
@@ -544,50 +544,44 @@ public void run(Reactor session) throws Exception {
...
-                                    activePlugins.remove(p);
-                                    plugins.remove(p);
-                                    LOGGER.log(Level.SEVERE, "Failed to install {0}: {1}", new Object[] { p.getShortName(), e.getMessage() });
-                                } catch (IOException e) {
-                                    failedPlugins.add(new FailedPlugin(p.getShortName(), e));
-                                    activePlugins.remove(p);
-                                    plugins.remove(p);
-                                    throw e;
-                                }
+                        g.followedBy().notFatal().attains(PLUGINS_PREPARED).add(String.format("Loading plugin %s v%s (%s)", p.getLongName(), p.getVersion(), p.getShortName()), session13 -> {
```
[the remark this answers is not in the document]
Comment (res0nance) [reply core/src/main/java/hudson/PluginManager.java:547]: `session` is already in scope

File: core/src/main/java/hudson/PluginManager.java [hunk 2]
```diff
@@ -544,50 +544,44 @@ public void run(Reactor session) throws Exception {
...
-                                }
-                                try {
-                                    p.getPluginOrFail().postInitialize();
-                                } catch (Exception e) {
-                                    failedPlugins.add(new FailedPlugin(p.getShortName(), e));
-                                    activePlugins.remove(p);
-                                    plugins.remove(p);
-                                    throw e;
-                                }
+                        g.followedBy().notFatal().attains(PLUGINS_STARTED).add("Initializing plugin " + p.getShortName(), session12 -> {
```
[the remark this answers is not in the document]
Comment (res0nance) [reply core/src/main/java/hudson/PluginManager.java:567]: `session` is already in scope

File: core/src/main/java/hudson/model/Executor.java [hunk 3]
```diff
@@ -344,38 +344,35 @@ public void run() {
...
+                    resetWorkUnit("was removed before the task's worker thread was ready to execute");
+                    return null;
+                }
+                // after this point we cannot unwind the assignment of the work unit, if the owner
+                // is removed or goes off-line then the build will just have to fail.
+                workUnit.setExecutor(Executor.this);
+                queue.onStartExecuting(Executor.this);
+                if (LOGGER.isLoggable(FINE))
+                    LOGGER.log(FINE, getName()+" grabbed "+workUnit+" from queue");
+                SubTask task1 = workUnit.work;
```
[the remark this answers is not in the document]
Comment (res0nance) [reply core/src/main/java/hudson/model/Executor.java:362]: `task` is already in scope

**Review by basil — APPROVED** [review]

**Comment by basil** [comment]
Suggest closing in favor of #5701.

**Pull request closed without merging**